d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -154,6 +176,31 @@ public void listenToService(String serviceName)
     });
   }
 
+  private boolean shouldNotListenToSymlink(String symlinkName, String fullResourceName)
```

Comment:
A method called `shouldNotListenToSymlink` should not have side-effects (like registering the watcher).